extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/ResteasyReactiveRecorder.java

```diff
@@ -225,4 +228,42 @@ public ServerRestHandler get() {
         }
     }
 
+    private static class MultipartHandlerSupplier implements Supplier<ServerRestHandler> {
+
+        private volatile Boolean handleFileUploads;
+        private volatile String uploadsDirectory;
+        private volatile Boolean mergeFormAttributes;
+        private volatile Boolean deleteUploadedFilesOnEnd;
+        private volatile Long maxBodySize;
+
+        @Override
+        public ServerRestHandler get() {
+            if (handleFileUploads == null) {
+                synchronized (this) {
+                    if (handleFileUploads == null) {
+                        // read the config like this in order to be able to keep ResteasyReactiveRecorder#createDeployment
+                        // as a static-init recorder
+                        // TODO: reevaluate this
+                        handleFileUploads = ConfigProvider.getConfig().getOptionalValue("quarkus.http.body.handle-file-uploads",
+                                Boolean.class).orElse(true);
+                        uploadsDirectory = ConfigProvider.getConfig().getOptionalValue("quarkus.http.body.uploads-directory",
+                                String.class).orElse("file-uploads");
+                        mergeFormAttributes = ConfigProvider.getConfig()
+                                .getOptionalValue("quarkus.http.body.merge-form-attributes",
+                                        Boolean.class)
+                                .orElse(true);
+                        deleteUploadedFilesOnEnd = ConfigProvider.getConfig()
+                                .getOptionalValue("quarkus.http.body.delete-uploaded-files-on-end", Boolean.class)
+                                .orElse(false);
```

Comment:
I'm not sure this should be an option, but at the very least it should default to `true` and force people to decide what to do with the uploads: move them someplace they take care of them, like dedicated FS, or cloud storage or DB, whatever, but away from our responsibility. Otherwise these will persist on application restart and grow forever without the user noticing.
IMO it's better for users to notice that we removed their files right away during dev (and document it) and decide what they need.

Reply:
Well, we already have this option in Quarkus so I just reused it.

I'm all for making this option `true` be default in Quarkus

Reply:
Oh, I didn't realise that. I would say yes, but it's possible that existing users depend on this behaviour then. I don't think we should follow that behaviour in RR.

Reply:
👍🏼